abstractions/php/src/AuthenticationProvider.php

```diff
@@ -0,0 +1,8 @@
+<?php
+namespace Microsoft\Kiota\Abstractions;
+
+use Http\Promise\Promise;
+
+interface AuthenticationProvider {
+    public function getAuthorizationToken(string $requestUrl): Promise;
```

Comment:
as far as I can tell PhP doesn't support declaring generics, if this is correct, no work needed, otherwise, probably need to type to Promise<string> https://wiki.php.net/rfc/generics#:~:text=Generic%20type%20relationships%20are%20already%20possible%20in%20PHP%2C,can%27t%20even%20be%20documented%2C%20e.g.%20using%20php-doc%20tags.